src/lib/controllers/DevicesController.js

```diff
@@ -112,17 +118,18 @@ class DevicesController extends Controller {
     postBody: Object,
   ): Promise<*> {
     try {
+      const userID = this.user.id;
       const result = await this._deviceRepository.callFunction(
         deviceID,
+        userID,
         functionName,
         postBody,
       );
 
-      // TODO add userID checking
-      const device = await this._deviceRepository.getByID(deviceID);
+      const device = await this._deviceRepository.getByID(deviceID, userID);
       return this.ok(deviceToAPI(device, result));
     } catch (error) {
-      if (error.indexOf('Unknown Function') >= 0) {
+      if (error.indexOf && error.indexOf('Unknown Function') >= 0) {
```

Comment:
Instead of doing this, change the thrown exception to be an Error instead of a string.